src/gui/qgsdoublevalidator.h

```diff
@@ -0,0 +1,149 @@
+/***************************************************************************
+                         qgsdoublevalidator.h  -  description
+                             -------------------
+    begin                : June 2020
+    copyright            : (C) 2020 by Sebastien Peillet
+    email                : sebastien.peillet@oslandia.com
+
+  adapted version of Qgslonglongvalidator + QgsFieldValidator
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSDOUBLEVALIDATOR_H
+#define QGSDOUBLEVALIDATOR_H
+
+#include <limits>
+#include <QRegExpValidator>
+#include <QLocale>
+#include "qgis_gui.h"
+#include "qgis_sip.h"
+
+/**
+ * \ingroup gui
+ *
+ * QgsDoubleValidator is a QLineEdit Validator that combines QDoubleValidator
+ * and QRegularExpressionValidator to allow user to enter double with both
+ * local and C interpretation as a fallback.
+ *
+ * \since QGIS 3.14
+ */
+class GUI_EXPORT QgsDoubleValidator : public QRegularExpressionValidator
+{
+    Q_OBJECT
+
+  public:
+
+    /**
+     * Constructor for QgsDoubleValidator.
+     */
+    explicit QgsDoubleValidator( QObject *parent );
+
+    /**
+     * Constructor for QgsDoubleValidator.
+     *
+     * bottom is the minimal range limit accepted by the validator
```

Comment:
```suggestion
     * \param bottom the minimal range limit accepted by the validator
```
can you please use the `\param` sintax for arguments? (here and everywhere)

Reply:
My bad... I add it !